Correct cache behavior for multiple ONS name lookups.

saravananp-001 opened the pull request:
When multiple ONS names are provided in the lookup, the data is consistently updated in cache using the first name_hash instead of being appended with a new name_hash each time.

saravananp-001 committed:
Correct cache behavior for multiple ONS name lookups.

When multiple ONS names are provided in the lookup, the data is consistently updated in cache using the first name_hash instead of being appended with a new name_hash each time.

darcys22 commented:
Thanks for the PR!

Is this compiling? Seeing this error in the CI
```
/drone/src/src/simplewallet/simplewallet.cpp: In member function 'bool cryptonote::simple_wallet::ons_lookup(std::vector<std::__cxx11::basic_string<char> >)':

/drone/src/src/simplewallet/simplewallet.cpp:7199:25: error: 'const class nlohmann::basic_json<>' has no member named 'name_hash'

 7199 |                 mapping.name_hash};
 ```
 
 I suspect the difference is probably just needing to be `mapping["name_hash"]` as the json struct behaves differently to a regular c++ struct

saravananp-001 committed:
compilation fix for mapping name_hash

error: 'const class nlohmann::basic_json<>' has no member named 'name_hash'

 7199 |                 mapping.name_hash};

resolved this error by replacing  mapping.name_hash with  mapping["name_hash"]

victor-tucci commented:
I updated the changes.
